streams/src/main/java/org/apache/kafka/streams/processor/TopologyBuilder.java

```diff
@@ -708,6 +814,22 @@ private void connectStateStoreNameToSourceTopics(final String stateStoreName,
                 Collections.unmodifiableSet(sourceTopics));
     }
 
+
+    private <T> void maybeAddToResetList(Collection<T> earliestResets, Collection<T> latestResets, AutoOffsetReset offsetReset, T item) {
+        if (offsetReset != null) {
```

Comment:
This check sounds fine to me, as this function is named `maybeAdd...` anyways :P